### PR TITLE
Add support for DDS

### PR DESCRIPTION
**Pull request by isatsam**
This adds support for DirectDraw Surface images.

The sample is [this image](https://imgur.com/a/ks8GDud) sourced from [sembiance format samples resource](https://sembiance.com/fileFormatSamples/image/dds/).

### CODE REVIEW

File: filetype/types/image.py [hunk 1]
```diff
@@ -402,3 +402,24 @@ def match(self, buf):
...
+            mime=Dds.MIME,
+            extension=Dds.EXTENSION
+        )
+
+    def match(self, buf):
+        return (len(buf) > 4 and
+                buf[0] == 0x44 and
+                buf[1] == 0x44 and
+                buf[2] == 0x53 and
+                buf[3] == 0x20)
```
Comment (babenek) [inline filetype/types/image.py:425]:
```suggestion
        return buf.startswith(b"\x44\x44\x53\x20")
```

Comment (isatsam) [reply filetype/types/image.py:425]: Looks a bit inconsistent with other match() methods in the same file, such as in Jpeg, Gif, Png etc...

Comment (babenek) [reply filetype/types/image.py:425]: ... but fasters. Moreover, the approach is used in the repo https://github.com/h2non/filetype.py/blob/590dac5736e67882fec4344cf39343de49a3407e/filetype/types/video.py#L70
I agree with specific bytes comparison when there is "random" order.

Comment (isatsam) [reply filetype/types/image.py]: I see I see, thanks!

**Comment by h2non** [comment]
The fixture file is too big, please make it under 5KB.

**Comment by isatsam** [comment]
Hi there, I've replaced the file with a ~3KB one. 
I haven't touched it in a bit and re-ran the tests now to check everything, it now produces a "No module named 'distutils'" error at the end, even if `setuptools` is installed - is that good to ignore (I'm guessing it's probably not related to what this PR changes)?

**Pull request merged**